### DISCUSSION TITLE
Error after adding mods to forge 1.19.2

### DISCUSSION THREAD

**Discussion by IniJamiel** [itzg/docker-minecraft-server#2269] [Q&A]
I made a docker container with TYPE=FORGE and VERSION=1.19.2 I have tried to use volumes to copy the mods, as well as copying the mods to /data/mods manually. But, I still get this error regardless of versions. Is there any steps that I miss? using curse modpacks seems to work in this environment. Running forge server without mods seems to work just fine
```
Exception in thread "main" cpw.mods.niofs.union.UnionFileSystem$UncheckedIOException: java.util.zip.ZipException: zip END header not found
Caused by: java.util.zip.ZipException: zip END header not found
```

**Comment by itzg** [comment]
It sounds like the jar/zip file is corrupted or not actually a zip file. I know sometimes if a download fails, it'll put the failure page HTML in the file instead of the expected binary content.

**Reply by 06NinjaKid06** [reply]
im having the same problem but i checked all the files with 7-zip and they're all fine

**Comment by damian123xD** [comment]
i have the same problem, any idea how to solve it?

**Reply by tendonut** [reply]
I'm kinda stuck in the same place. Using ATM9. MC ver. 1.20.1. There are over 400 mods installed, and the error doesn't really indicate which one may be causing a problem (and it took me 6 tries and 2 hours to download them all because of the CurseForge rate limit with API calls)

**Reply by itzg** [reply]
That is ideally a problem for the modpack author to solve. The only advice I can provide with respect to this image is to make sure you do **not** have a restart policy enabled. Otherwise you will hit the API rate limit more quickly.

**Reply by tendonut** [reply]
I was able to do a `
find $PATH_TO_MODS -name "*jar" | xargs -L 1 zip -T | grep error | grep invalid`
to find out which jars were corrupted (it was like....a third of them), delete them, then restart the container to have it re-download them. Took about 3 loops of that to finally get all of them clean. Still have the error, but I am heading in the right direction. I won't enable the restart-policy until I got this fully working. 

The issue seems to be CurseForge downloads getting corrupted during the download phase, not even the modpack publisher.

**Reply by GamingAlex553** [reply]
Try to reinstal, maybe is a coruption. Same thing happen to me with Sinytra Connector, giving the same error and that mod work, after I reinstall it.

**Comment by psoloviov** [comment]
Hey everyone, not sure that this issue can be related to forge issues, but please, check what I wrote in different issue:
https://github.com/itzg/docker-minecraft-server/issues/2283#issuecomment-2907712676

In my case, the problem was not with mods, but with forge and their bad installation.

**Reply by GamingAlex553** [reply]
Maybe.
